Yosemite/Yosemite/Stores/ReceiptStore.swift

```diff
@@ -56,4 +62,66 @@ private extension ReceiptStore {
         let renderer = ReceiptRenderer(content: content)
         onContent(renderer.htmlContent())
     }
+
+    func loadReceipt(order: Order, onCompletion: @escaping (Result<CardPresentReceiptParameters, Error>) -> Void) {
+
+        guard let outputURL = try? fileURL(order: order) else {
+            let error = ReceiptStoreError.fileNotFound
+            onCompletion(.failure(error))
+            return
+        }
+
+        guard let receiptContent: ReceiptContent = try? fileStorage.data(for: outputURL) else {
+            DDLogError("⛔️ Unable to load receipt metadata for order: \(order.orderID)")
+            let error = ReceiptStoreError.fileError
+            onCompletion(.failure(error))
+
+            return
+        }
+
+        onCompletion(.success(receiptContent.parameters))
+    }
+
+    func saveReceipt(order: Order, parameters: CardPresentReceiptParameters) {
+        let lineItems = order.items.map { ReceiptLineItem(title: $0.name, quantity: $0.quantity.description, amount: $0.price.stringValue)}
+
+        let content = ReceiptContent(parameters: parameters, lineItems: lineItems)
+
+        guard let outputURL = try? fileURL(order: order) else {
+            DDLogError("⛔️ Unable to create file for receipt for order id: \(order.orderID)")
+
+            return
+        }
+
+
+        do {
+            try fileStorage.write(content, to: outputURL)
+            Swift.print("new receipt saved: open \(outputURL.path)") // command to open the generated file
+        } catch {
+            DDLogError("⛔️ Unable to save receipt for order id: \(order.orderID)")
+        }
+    }
+}
+
+private extension ReceiptStore {
+    func fileURL(order: Order) throws -> URL {
+        return try FileManager.default.url(for: .documentDirectory,
+                                                           in: .userDomainMask,
+                                                           appropriateFor: nil,
+                                                           create: false)
+                .appendingPathComponent(fileName(order: order))
+                .appendingPathExtension("plist")
+    }
+
+    func fileName(order: Order) -> String {
+        "order-id-\(order.orderID)-receipt"
```

Comment:
This will be fine I think for now, but we should probably support multiple receipts in the future. For example, if a in-person card payment buyer gets a refund, they usually ask for a receipt for that too.

Reply:
True, good point

Reply:
Oh, another thing to keep in mind is multiple stores, and more generally when we should remove these files from the device.

Reply:
Yep. I can add the storeid to the file name as well. Removing the files... as soon as we have an endpoint that we can use to fetch this data